hot.h

```diff
@@ -0,0 +1,297 @@
+#ifndef INCLUDED_OPENTRACING_HOT_H
+#define INCLUDED_OPENTRACING_HOT_H
+
+// -------------- Copied from noop.h --------------
+
+#include <opentracing/baggage.h>
+#include <opentracing/span.h>
+#include <opentracing/spancontext.h>
+#include <opentracing/spanoptions.h>
+#include <opentracing/stringref.h>
+#include <opentracing/tracer.h>
+#include <stdint.h>
+
+namespace opentracing {
+
+template <typename T1, typename T2> class HotAdapter;
+template <typename T1, typename T2> class HotContext;
+template <typename T1, typename T2> class HotOptions;
+template <typename T1, typename T2> class HotSpan;
+template <typename T1, typename T2> class HotTracer;
+
+template <typename T1, typename T2>
+class HotAdapter {
+    typedef HotContext<T1, T2> MyHotContext;
+  public:
+    typedef MyHotContext*       iterator;
+    typedef const MyHotContext* const_iterator;
+
+    BaggageRef ref(const const_iterator& it) const
+    {
+        static const char empty[] = "";
+        return BaggageRef(empty, empty);
+    }
+
+    Baggage copy(const const_iterator& it) const
+    {
+        return Baggage("", "");
+    }
+};
+
+
+/**
+ * Note: VERY TRICKY. Pay attention to const-ness here. Span's context mustnn't
+ * be const but `extract()`'s has to be const. To be properly thought/implemented.
+ */
+template <typename T1, typename T2>
+class HotContext : public GenericSpanContext<HotContext<T1, T2>, HotAdapter<T1, T2> > {
+    typedef typename T1::SpanContext SpanContext1;
+    typedef typename T2::SpanContext SpanContext2;
+
+    const SpanContext1 * ctx1; // Note: Might be NULL
+    const SpanContext2 * ctx2; // Note: Might be NULL
+
+    friend class HotTracer<T1, T2>;
+    friend class HotSpan<T1, T2>;
+
+    HotContext(const SpanContext1 * ctx1_, const SpanContext2 * ctx2_)
+        : ctx1(ctx1_)
+        , ctx2(ctx2_)
+    {
+    }
+
+    HotContext()
+    {
+    }
+
+  public:
+
+    typedef GenericSpanContext<HotContext<T1, T2>, HotAdapter<T1, T2> > BaggageIterator;
+
+    BaggageIterator baggageBeginImp() const
+    {
+        return baggageEndImp();
+    }
+    BaggageIterator baggageEndImp() const
+    {
+        return BaggageIterator(this);
+    }
+
+    int getBaggageImp(const StringRef&, std::string*) const
```

Comment:
Here we have the usual `getBaggage()` issue mentioned in https://github.com/opentracing/specification/issues/9#issuecomment-270208111